centrallix/osdrivers/objdrv_fp.c

```diff
@@ -1243,18 +1244,15 @@ fp_internal_ParseColumn(pFpColInf column, pObjData pod, char* data, char* row_da
 		if (column->DecimalOffset) pod->Double /= pow(10, column->DecimalOffset);
 		break;
 	    case DATA_T_MONEY:
+            //decimalOffsetValue is originally 10000 to convert v to 10000ths of a dollar
+            //decimalOffsetValue is divided by 10, column->DecimalOffset times,
+            //keeping the decimal as a double in case it drops below 0
+            //Finally, I multiple v by decimalOffsetValue to get my Money->Value
 		if (fp_internal_MappedCopy(ibuf, sizeof(ibuf), column, row_data) < 0) return -1;
 		v = strtoll(ibuf, NULL, 10);
-		f = 1;
-		for(i=0;i<column->DecimalOffset;i++) f *= 10;
+		for (i=0; i<column->DecimalOffset; i++) decimalOffsetValue /= 10.0;
 		pod->Money = (pMoneyType)data;
-		pod->Money->WholePart = v/f;
-		v = (v/f)*f;
-		if (column->DecimalOffset <= 4)
-		    for(i=column->DecimalOffset;i<4;i++) v *= 10;
-		else
-		    for(i=4;i<column->DecimalOffset;i++) v /= 10;
-		pod->Money->FractionPart = v;
+		pod->Money->Value = v*decimalOffsetValue;
```

Comment:
The challenge here is that this cannot be straightforwardly tested due to no real way to do column type autodetection - all columns are tagged as DATA_T_STRING.  So even with test data we can't trigger this segment of code without handcrafting the data structures.

I would recommend using pow() instead of the repeated division, as pow() may have a better way of managing error as opposed to the cumulative error from repeated division on a floating point type.  Also, when assigning to ->Value, add 0.1 to the product of v*decimalOffsetValue, also dealing with potential floating point error (like 10.0 becoming 9.999999999999 and then truncating to 9 for the long long assignment).

Reply:
So am I correctly understanding you that this code would never actually run in a real environment - i.e. unless you call this internal function directly with handcrafted data - since columns will always just be tagged as DATA_T_STRING? If so, could you clarify why all this code still exists for other data types and why it should be updated for the new MoneyType, if it's not being used? Are there plans to implement column type autodetection sometime, or is this code never going to be used and should maybe just be removed...?

Reply:
Correct, it would never run in a production environment.  If y'all would prefer to remove the non-DATA_T_STRING cases, or at least #if 00 them out, you may go ahead (though I'm not concerned about its presence) _after_ we merge this branch into master, so the work done here is in the permanent history and we can get it back & test it at a later date if needed.

Reply:
Okay, thanks for clarifying. It would be great to talk a bit more sometime about the design rationale behind that approach, but that can wait :)

Reply:
The main thing to me is to ensure the modified code in question doesn't get squashed out in the merge, and to try to separate the money type work from the removal of this unreachable section of code.